### DISCUSSION TITLE
Running F Prime Application Natively

### DISCUSSION THREAD

**Discussion by CosmoBytes** [nasa/fprime#3004] [Q&A]
Greetings, everyone!

I’m new to the F Prime framework and have successfully completed the tutorial, which went smoothly. Now, I'm eager to explore how to install and run the application without relying on `fprime-gds` or a virtual environment. My goal is to have it run natively on a Raspberry Pi (or a similar device) and communicate over Wi-Fi with another device running ground station software.

In my setup, I generated the `HelloWorldDeployment`, which created two directories: `build-artifacts` and `build-fprime-automatic-native`. Each contains its own binary for the application. When I run the binary with the following command:

```bash
./HelloWorldDeployment -a 0.0.0.0 -p 7000
```

I encounter the following warning:

```
[WARNING] Failed to open port with status -4 and errno 111
```

I selected the TCP server option during setup, so I expected the application to act as the server. 

**My questions are:**

1. Is there something I might have missed in the setup?
2. How can I run this application from the provided binary without `fprime-gds`?

Any insights or guidance would be greatly appreciated!

Thank you!

**Comment by Joshua-Anderson** [comment]
This looks like the correct way to launch a standard fprime application to me! From my googling, errno 111 is connection refused. Are you sure you setup your hello world deployment as a TCP server instead of a TCP client?
